test/DelimitedDataParser.Test/ParserTest.Reader.cs

```diff
@@ -1892,5 +1893,87 @@ public void ParseReader_HasRows_ReturnsTrueForHeaderAndDataRows()
 
             Assert.True(reader.HasRows);
         }
+
+        [Fact]
+        public void Supports_Varying_Column_Counts_In_Enumerator_With_Column_Headers()
+        {
+            var content = @"Field 1,Field 2,Field 3" + Environment.NewLine
+                + @"Data 1,Data 2" + Environment.NewLine
+                + @"Data 5" + Environment.NewLine
+                + @"Data 6,Data 7,Data 8, data 9";
+
+            using (var stringReader = new StringReader(content))
+            {
+                var parser = new Parser
+                {
+                    UseFirstRowAsColumnHeaders = true
+                };
+
+                var reader = parser.ParseReader(stringReader);
+                var enumerator = reader.GetEnumerator();
+
+                enumerator.MoveNext();
+                var current = enumerator.Current as IDataRecord;
+                Assert.Equal("Data 1", current[0]);
+                Assert.Equal("Data 2", current[1]);
+                Assert.Null(current[2]);
+
+                enumerator.MoveNext();
+                current = enumerator.Current as IDataRecord;
+                Assert.Equal("Data 5", current[0]);
+                Assert.Null(current[1]);
+                Assert.Null(current[2]);
+
+                enumerator.MoveNext();
+                current = enumerator.Current as IDataRecord;
+                Assert.Equal("Data 6", current[0]);
+                Assert.Equal("Data 7", current[1]);
+                Assert.Equal("Data 8", current[2]);
+            }
+        }
+
+        [Fact]
+        public void Supports_Varying_Column_Counts_In_Enumerator_Without_Column_Headers()
+        {
+            var content = @"Field 1,Field 2,Field 3" + Environment.NewLine
+                + @"Data 1,Data 2" + Environment.NewLine
+                + @"Data 5" + Environment.NewLine
+                + @"Data 6,Data 7,Data 8, data 9";
```

Comment:
Could we modify this test case to use the same example as provided in #22? We don't need three data rows to reproduce the reported issue.